openmdao/core/vec_wrapper.py

```diff
@@ -278,7 +278,7 @@ def _get_scalar_units_complex(self):
 
     def _set_arr(self, value):
         """Set an array value."""
-        self.val[:] = value.flat
+        self.val[:] = value.real.flat
```

Comment:
Adding a call to .real to every vecwrapper set call will have some negative performance impact.  Probably not that large, but it would still be better to track down where this complex value is being passed in so that we could do .real there instead of in every set call.